go/v1beta1/storage/storage_test.go

```diff
@@ -0,0 +1,886 @@
+	t.Run("CreateProject", func(t *testing.T) {
+		_, gp, cleanUp := createStore(t)
+		defer cleanUp()
+
+		ctx := context.Background()
+		pID := "project"
+		p := &prpb.Project{}
+		p.Name = name.FormatProject(pID)
+		if _, err := gp.CreateProject(ctx, pID, p); err != nil {
+			t.Errorf("CreateProject got %v want success", err)
+		}
+		// Try to insert the same project twice, expect failure.
+		if _, err := gp.CreateProject(ctx, pID, p); err == nil {
+			t.Errorf("CreateProject got success, want Error")
+		} else if s, _ := status.FromError(err); s.Code() != codes.AlreadyExists {
+			t.Errorf("CreateProject got code %v want %v", s.Code(), codes.AlreadyExists)
+		}
+	})
+
+	t.Run("CreateNote", func(t *testing.T) {
+		g, gp, cleanUp := createStore(t)
+		defer cleanUp()
+
+		ctx := context.Background()
+		nPID := "vulnerability-scanner-a"
+		if _, err := gp.CreateProject(ctx, nPID, &prpb.Project{}); err != nil {
+			t.Errorf("CreateProject got %v want success", err)
+		}
+
+		n := TestNote(nPID)
+		if _, err := g.CreateNote(ctx, nPID, TestNoteID, "userID", n); err != nil {
+			t.Errorf("CreateNote got %v want success", err)
+		}
+		// Try to insert the same note twice, expect failure.
+		if _, err := g.CreateNote(ctx, nPID, TestNoteID, "userID", n); err == nil {
+			t.Errorf("CreateNote got success, want Error")
+		} else if s, _ := status.FromError(err); s.Code() != codes.AlreadyExists {
+			t.Errorf("CreateNote got code %v want %v", s.Code(), codes.AlreadyExists)
+		}
+	})
+
+	t.Run("CreateOccurrence", func(t *testing.T) {
+		g, gp, cleanUp := createStore(t)
+		defer cleanUp()
+
+		ctx := context.Background()
+		nPID := "vulnerability-scanner-a"
+		if _, err := gp.CreateProject(ctx, nPID, &prpb.Project{}); err != nil {
+			t.Errorf("CreateProject got %v want success", err)
+		}
+
+		n := TestNote(nPID)
+		if _, err := g.CreateNote(ctx, nPID, TestNoteID, "userID", n); err != nil {
+			t.Errorf("CreateNote got %v want success", err)
+		}
+
+		oPID := "occurrence-project"
+		o := TestOccurrence(oPID, n.Name)
+		if _, err := g.CreateOccurrence(ctx, nPID, "userID", o); err != nil {
+			t.Errorf("CreateOccurrence got %v want success", err)
+		}
+		// Try to insert the same occurrence twice, expect failure.
+		if _, err := g.CreateOccurrence(ctx, nPID, "userID", o); err == nil {
+			t.Errorf("CreateOccurrence got success, want Error")
+		} else if s, _ := status.FromError(err); s.Code() != codes.AlreadyExists {
+			t.Errorf("CreateOccurrence got code %v want %v", s.Code(), codes.AlreadyExists)
+		}
+
+		pID, oID, err := name.ParseOccurrence(o.Name)
+		if err != nil {
+			t.Fatalf("Error parsing projectID and occurrenceID %v", err)
+		}
+
+		got, err := g.GetOccurrence(ctx, pID, oID)
+		if err != nil {
+			t.Fatalf("GetOccurrence got %v, want success", err)
+		}
+		opt := cmp.FilterPath(func(p cmp.Path) bool { return p.String() == "CreateTime" }, cmp.Ignore())
```

Comment:
Maybe a test that create time exists and is reasonable? If you want exact timestamps you could use the clockwork package, but making sure the timestamp is > the start of the test is probably good enough.

Reply:
That's a good idea, but I think it's out of scope for this PR. I added a TODO to address this in the future.